funcx_endpoint/funcx_endpoint/executors/high_throughput/interchange.py

```diff
@@ -1198,12 +1198,21 @@ def starter(comm_q: mp.Queue, *args, **kwargs) -> None:
     The executor is expected to call this function. The args, kwargs match that of the
     Interchange.__init__
     """
-    ic = Interchange(*args, **kwargs)
-    comm_q.put((ic.worker_task_port, ic.worker_result_port))
-    comm_q.close()
-    comm_q.join_thread()
-    del comm_q
-    ic.start()
+    ic = None
+    try:
+        ic = Interchange(*args, **kwargs)
+        comm_q.put((ic.worker_task_port, ic.worker_result_port))
+    finally:
+        if not ic:  # There was an exception
+            comm_q.put(None)
```

Comment:
I didn't know what .put(None) did... will have to look that up.

Reply:
The executor waits on the comm_q, and putting in a `None` lets the executor know that the Interchange failed to start. The executor is waiting for a non-null value to confirm Interchange start.

Reply:
FWIW, the general concept is "poison pill."  `None` is nothing special other than "an item in the queue that the other end will recognize as a signal to quit."